src/modules/form/util/formUtil.ts

```diff
@@ -323,11 +324,17 @@ const evaluateEnableWhen = ({
       en.compareQuestion ? values[en.compareQuestion] : undefined,
     ].filter((e) => !isNil(e))[0];
   }
-
+  if (isDate(currentValue)) {
+    comparisonValue = parseHmisDateString(comparisonValue);
```

Comment:
I'm not sure about changing the pattern here. In other places, like `ValueBound`, if we're comparing with a date then we use the date-typed `value_date` field.

With this change, we would be using the `answer_code` (presumably?) field to store the comparison date on the `EnableWhen`  object, rather than adding a typed `answer_date` field to the schema. Do we have a reason for that inconsistency?

Reply:
Good point, no reason, we can update the schema instead.

This was sort of a sidebar bugfix that doesn't have its own ticket, but I can create a ticket now or just proceed until this is ready to merge, whatever you prefer

Reply:
Ok, Lets add a separate ticket since its new functionality